criu/files.c

```diff
@@ -831,10 +833,35 @@ static void collect_desc_fle(struct fdinfo_list_entry *new_le, struct file_desc
 	}
 }
 
+static int add_pid_to_dmabuf_list(int pid)
+{
+	struct fdinfo_list_entry *le;
+
+	list_for_each_entry(le, &dmabuf_processes, desc_list)
+		if (le->pid == pid)
+			return 0;
+
+	le = alloc_fle(pid, NULL);
+
+	if (!le)
+		return -ENOMEM;
+
+	list_add(&le->desc_list, &dmabuf_processes);
+
+	return 0;
+}
+
 struct fdinfo_list_entry *collect_fd_to(int pid, FdinfoEntry *e, struct rst_info *rst_info, struct file_desc *fdesc,
 					bool fake, bool force_master)
 {
 	struct fdinfo_list_entry *new_le;
+	int ret;
+
+	if (fdesc->ops->type == FD_TYPES__EXT) {
+		ret = add_pid_to_dmabuf_list(pid);
```

Comment:
can we introduce a new plugin hook `COLLECT_FD` and move all other logic about dmabuf into the amd plugin?